### PR TITLE
Simulastsharedtask

### PR DESCRIPTION
**Pull request by xutaima**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [ ] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes # (issue).

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: examples/simultaneous_translation/docs/baseline.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# **Baseline Simultaneous Translation**
+---
+
+This is an instruction of training and evaluating a *wait-k* simultanoes LSTM model on MUST-C English-Gernam Dataset.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/baseline.md]:
```suggestion
This is an instruction of training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# Introduction to evaluation interface
+The simultaneous translation models from sharedtask participents are evaluated under a server-client protocol. The participents are requisted to plug in their own model API in the protocol, and submit a docker file.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
The simultaneous translation models from the shared task participants are evaluated under a server-client protocol. The participants are required to plug in their own model API in the protocol, and submit a Docker image.
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+## Server-Client Protocol
+An server-client protocol that will be used in evaluation. For example, when a *wait-k* model (k=3) translate the English sentence "Alice and Bob are good friends" to Genman sentence "Alice und Bob sind gute Freunde." , the evaluation process is shown as following figure.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
A server-client protocol will be used in evaluation. For example, when a *wait-k* model (k=3) translate the English sentence "Alice and Bob are good friends" to the German sentence "Alice und Bob sind gute Freunde." , the evaluation process is shown in the following figure. 
```

Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]: Where is the figure?

File: examples/simultaneous_translation/docs/baseline.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
+# **Baseline Simultaneous Translation**
+---
+
+This is an instruction of training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/baseline.md]:
```suggestion
You can find here instructions for training and evaluating a *wait-k* simultaneous LSTM model on the MUST-C English-German dataset.
```

File: examples/simultaneous_translation/docs/baseline.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    --log-format json \
+    --log-interval 10 \
+    --criterion cross_entropy_acc \
+    --user-dir $FAIRSEQ/examples/simultaneous_translation
+```
+
+## Evaluation
+---
+### Evaluation Server
+For text translation models, the server is set up as follow give input file and reference file.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/baseline.md]:
```suggestion
For text translation models, the server needs an input file and reference file. 
```

File: examples/simultaneous_translation/docs/baseline.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+``` shell
+python ./eval/server.py \
+    --hostname localhost \
+    --port 12321 \
+    --ref-file $DATA_ROOT \
+    --data-type speech
+```
+
+### Decode and Evaluate with Client
+Once the server is set up, run client to evaluate translation quality and latency.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/baseline.md]:
```suggestion
Once the server is set up, run the client to evaluate translation quality and latency.
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
+# Introduction to evaluation interface
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
# Introduction to the evaluation interface
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+The simultaneous translation models from the shared task participants are evaluated under a server-client protocol. The participants are required to plug in their own model API in the protocol, and submit a Docker image.
+
+## Server-Client Protocol
+A server-client protocol will be used in evaluation. For example, when a *wait-k* model (k=3) translate the English sentence "Alice and Bob are good friends" to the German sentence "Alice und Bob sind gute Freunde." , the evaluation process is shown in the following figure. 
+
+While every time client needs to read a new state (word or speech utterence), a "GET" request is supposed to sent over to server. Whenever a new token is generated, a "SEND" request with the word predicted (untokenized word) will be sent to server immediately. The server can hence calculate both latency and BLEU score of the sentence.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
Every time the client needs to read a new state (word or speech chunk), a "GET" request needs to be sent to the server. Whenever a new token is generated, a "SEND" request with the predicted word (untokenized word) will be sent to the server immediately. The server can hence calculate both the latency and BLEU score of the sentence.
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+### Server
+The server code is provided and can be set up directly locally for development purpose. For example, to evaluate a text simultaneous test set,
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
The server code is provided and can be set up locally for development purposes. For example, to evaluate a text simultaneous test set,
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+The client will handle the evaluation process mentioned above. It should be out-of-box as well. The client's protocol is as following table
+
+|Action|Content|
+|:---:|:---:|
+|Request new word / utterence| ```{key: "Get", value: None}```|
+|Predict word "W"| ```{key: "SEND", value: "W"}```|
+
+
+
+The core of the client module is the agent, which needs to be modified to different models accordingly. The abstract class of agent is as follow, the evaluation process happens in the `decode()` function.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
The core of the client module is the agent, which needs to be modified for different models accordingly. The abstract class of agent is as follow, the evaluation process happens in the `decode()` function. 
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+    def reset(self):
+        # Reset agent
+        ...
+        
+    def decode(self, session):
+        
+        states = self.init_states()
+        self.reset()      
+
+        # Evaluataion protocol happens here
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
        # Evaluation protocol happens here
```

File: examples/simultaneous_translation/docs/evaluation.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+## Quality
+The quality is measured by detokenized BLEU. So make sure that the predicted words sent to server are detokenized. An implementation is can be find [here](../eval/agent.py)
+
+## Latency
+The latency metrics are 
+* Average Proportion
+* Average Lagging
+* Differentiable Average Lagging
+Again Thery will also be evaluated on detokenized text.
```
Comment (jmp84) [inline examples/simultaneous_translation/docs/evaluation.md]:
```suggestion
Again, they will also be evaluated on detokenized text.
```
